packages/sdk/src/syncedStream.ts

```diff
@@ -21,31 +44,91 @@ export class SyncedStream extends Stream implements ISyncedStream {
     ) {
         super(userId, streamId, clientEmitter, logEmitFromStream)
         this.log = dlog('csb:syncedStream', { defaultEnabled: false }).extend(userId)
+        this.logError = dlogError('csb:syncedStream').extend(userId)
         this.persistenceStore = persistenceStore
+        this.metrics = {
+            initializationTime: 0,
+            lastAccessTime: Date.now(),
+            retryCount: 0,
+            memoryUsage: 0,
+            successfulOperations: 0,
+            failedOperations: 0,
+        }
     }
 
-    async initializeFromPersistence(persistedData?: LoadedStream): Promise<boolean> {
-        const loadedStream =
-            persistedData ?? (await this.persistenceStore.loadStream(this.streamId))
-        if (!loadedStream) {
-            this.log('No persisted data found for stream', this.streamId, persistedData)
-            return false
+    private calculateMemoryUsage(): number {
+        const streamSize = {
+            streamId: this.streamId,
+            userId: this.userId,
+            events: this.view.events.size,
+            timeline: this.view.timeline.length,
         }
+        return JSON.stringify(streamSize).length / 1024 // KB
```

Comment:
this is not how you calculate memory usage - shove some random stuff into a json blob and count the length of the generated string? All streams are going to be the same size.